config/config_test.go

```diff
@@ -180,6 +181,86 @@ func TestInitConfig(t *testing.T) {
 	assert.Equal(t, "", param.Federation_DiscoveryUrl.GetString())
 }
 
+func TestHomeDir(t *testing.T) {
+	if runtime.GOOS != "windows" {
```

Comment:
One final small change that just occurred to me. Instead of wrapping all this in the `if runtime.GOOS != "windows"`, can you do:
```
if runtime.GOOS == "windows" {
    t.Skip("Skipping test on Windows, as it's not expected to work")
}

// Rest of the function
```

This gets rid of one layer of nesting that makes the test a bit harder to read, and also prints a message when the test is run on Windows.